### ISSUE TITLE
Throttle API requests on keyup/keydown/keypress

### ISSUE THREAD

**Issue by artfuldev** [varnamproject/varnamproject.com#5]
When we type fast, a number of requests are made. We could throttle this by debouncing the events.

**Comment by navaneeth** [comment]
Yes. Lot of requests go waisted as you would have advanced in typing next
words. So the client checks this condition and discards the response. But as
you rightly observed, server will be processing unnecessary requests. We
need to cancel them if it is not sent.

The online editor was created merely to showcase varnam. That is the reason
why lot of these are not addressed.

Online editor is never expected to be the primary editor. Varnam is supposed to be used with native
keyboards and fully offline so that you get the speed that you need. But we

## have done only `Linux` IME so far. Windows and OSX are pending. 

Thanks
Navaneeth
